### PR TITLE
Convert BezelImageView, EndlessOnScrollListener, ComposeScheduleView, ProgressImageView to Kotlin

### PR DESCRIPTION
**Pull request by nikclayton**

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/components/compose/view/ComposeScheduleView.kt [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+        init(context)
+    }
+
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
+        context,
+        attrs,
+        defStyleAttr
+    ) {
+        init(context)
+    }
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/components/compose/view/ComposeScheduleView.kt:64]: please merge the constructors into one with default values like e.g. https://github.com/tuskyapp/Tusky/blob/develop/app/src/main/java/com/keylesspalace/tusky/components/compose/view/FocusIndicatorView.kt#L23

File: app/src/main/java/com/keylesspalace/tusky/components/compose/view/ComposeScheduleView.kt [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+            this
+        )
+        dateFormat = SimpleDateFormat.getDateInstance()
+        timeFormat = SimpleDateFormat.getTimeInstance()
+        iso8601 = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.getDefault())
+        iso8601.timeZone = TimeZone.getTimeZone("UTC")
+        binding.scheduledDateTime.setOnClickListener { openPickDateDialog() }
+        binding.invalidScheduleWarning.setText(R.string.warning_scheduling_interval)
+        scheduleDateTime = null
+        updateScheduleUi()
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/components/compose/view/ComposeScheduleView.kt:78]: Do the initialization of these right were they are defined please, there is no need for `lateinit var`

File: app/src/main/java/com/keylesspalace/tusky/components/compose/view/ComposeScheduleView.kt [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    }
+
+    private fun onTimeSet(hourOfDay: Int, minute: Int) {
+        initializeSuggestedTime()
+        scheduleDateTime?.set(Calendar.HOUR_OF_DAY, hourOfDay)
+        scheduleDateTime?.set(Calendar.MINUTE, minute)
+        updateScheduleUi()
+        if (listener != null) {
+            listener!!.onTimeSet(time)
+        }
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/components/compose/view/ComposeScheduleView.kt:207]:
```suggestion
        listener?.onTimeSet(time)
```

File: app/src/main/java/com/keylesspalace/tusky/view/BezelImageView.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ */
+class BezelImageView @JvmOverloads constructor(
+    context: Context?,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : BezelImageView(
+    context!!,
+    attrs,
+    defStyle
+) {
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/view/BezelImageView.kt:35]:
```suggestion
class BezelImageView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyle: Int = 0
) : BezelImageView(
    context,
    attrs,
    defStyle
) {
```

File: app/src/main/java/com/keylesspalace/tusky/components/compose/view/ComposeScheduleView.kt (hunk 1 above)
Comment (connyduck) [reply app/src/main/java/com/keylesspalace/tusky/components/compose/view/ComposeScheduleView.kt:64]: also in ProgressImageView

File: app/src/main/java/com/keylesspalace/tusky/components/compose/view/ComposeScheduleView.kt [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+    }
+
+    val time: String?
+        get() = if (scheduleDateTime == null) {
+            null
+        } else iso8601.format(scheduleDateTime!!.time)
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/components/compose/view/ComposeScheduleView.kt:213]:
```suggestion
        get() = scheduleDateTime?.time?.let { iso8601.format(it) }
```

File: app/src/main/java/com/keylesspalace/tusky/components/compose/view/ProgressImageView.kt [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        super.onDraw(canvas)
+        val angle = progress / 100f * 360 - 90
+        val halfWidth = width / 2f
+        val halfHeight = height / 2f
+        progressRect[halfWidth * 0.75f, halfHeight * 0.75f, halfWidth * 1.25f] = halfHeight * 1.25f
+        biggerRect.set(progressRect)
+        val margin = 8
+        biggerRect[progressRect.left - margin, progressRect.top - margin, progressRect.right + margin] =
+            progressRect.bottom + margin
+        canvas.saveLayer(biggerRect, null, Canvas.ALL_SAVE_FLAG)
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/components/compose/view/ProgressImageView.kt:95]: deprecated

File: app/src/main/java/com/keylesspalace/tusky/components/compose/view/ProgressImageView.kt [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    private fun init() {
+        circlePaint.color = context.getColor(R.color.tusky_blue)
+        circlePaint.strokeWidth = Utils.dpToPx(context, 4).toFloat()
+        circlePaint.style = Paint.Style.STROKE
+        clearPaint.xfermode = PorterDuffXfermode(PorterDuff.Mode.DST_OUT)
+        markBgPaint.style = Paint.Style.FILL
+        markBgPaint.color = context.getColor(R.color.tusky_grey_10)
+        captionDrawable = AppCompatResources.getDrawable(context, R.drawable.spellcheck)!!
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/components/compose/view/ProgressImageView.kt:66]: inline above

File: app/src/main/java/com/keylesspalace/tusky/components/compose/view/ProgressImageView.kt [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+        biggerRect[progressRect.left - margin, progressRect.top - margin, progressRect.right + margin] =
+            progressRect.bottom + margin
+        canvas.saveLayer(biggerRect, null, Canvas.ALL_SAVE_FLAG)
+        if (progress != -1) {
+            canvas.drawOval(progressRect, circlePaint)
+            canvas.drawArc(biggerRect, angle, 360 - angle - 90, true, clearPaint)
+        }
+        canvas.restore()
+        val circleRadius = Utils.dpToPx(context, 14)
+        val circleMargin = Utils.dpToPx(context, 14)
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/components/compose/view/ProgressImageView.kt:102]: Calling this twice inside onDraw despite it always retuning the same does not look like a good idea. How about some class member fourteenDpInPx and then a comment explaining it is used both as radius and as margin?
(`onDraw` should really be as minimal as possible for perf reasons, it is called very often)

File: app/src/main/java/com/keylesspalace/tusky/components/compose/view/ProgressImageView.kt [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+        val circleY = height - circleMargin - circleRadius / 2
+        val circleX = width - circleMargin - circleRadius / 2
+        canvas.drawCircle(circleX.toFloat(), circleY.toFloat(), circleRadius.toFloat(), markBgPaint)
+        captionDrawable.setBounds(
+            width - circleMargin - circleRadius,
+            height - circleMargin - circleRadius,
+            width - circleMargin,
+            height - circleMargin
+        )
+        captionDrawable.setTint(Color.WHITE)
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/components/compose/view/ProgressImageView.kt:112]: could be moved to initialization of the captionDrawable

**Review by connyduck — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/keylesspalace/tusky/components/compose/view/ComposeScheduleView.kt (hunk 1 above)
Comment (nikclayton) [reply app/src/main/java/com/keylesspalace/tusky/components/compose/view/ComposeScheduleView.kt:64]: Ah, neat trick.

**Comment by nikclayton** [comment]
Thanks for the feedback. Applied.

**Review by connyduck — APPROVED** [review]

**Pull request merged**